mlir/test/CMakeLists.txt

```diff
@@ -109,6 +110,7 @@ if(ROCMLIR_DRIVER_ENABLED)
     mlir-translate
     opt
     llc
+    fusion-e2e-tests
```

Comment:
Wrong list of dependencies. You want this handled the same as the `e2e` test directory.

Reply:
I'd move it into the conditional. Heck, I'd move the e2e fusion tests under `e2e/`, thus needing fewer lit configs.

Reply:
Turns out that making it conditional is also required to make -DBUILD_FAT_LIBROCKCOMPILER=ON work.